cmd/squid-per-site-exporter/main.go

```diff
@@ -0,0 +1,283 @@
+	squidBytesTotal.WithLabelValues(hostname).Add(float64(bytes))
+	squidResponseTime.WithLabelValues(hostname).Observe(elapsedTime / 1000.0) // Convert ms to seconds
+
+	if isHit {
+		squidHitTotal.WithLabelValues(hostname).Inc()
+	} else {
+		squidMissTotal.WithLabelValues(hostname).Inc()
+	}
+
+	// Ensure both hit and miss counters are initialized (even if 0) for this hostname
+	// This ensures squid_site_hits_total appears in metrics output even with 0 value
+	squidHitTotal.WithLabelValues(hostname).Add(0)
+	squidMissTotal.WithLabelValues(hostname).Add(0)
+
+	// Update hit ratio
+	if stats.Requests > 0 {
+		ratio := float64(stats.Hits) / float64(stats.Requests)
+		squidHitRatio.WithLabelValues(hostname).Set(ratio)
+	}
+}
+
+func (e *Exporter) readFromStdin() {
+	log.Printf("Reading squid logs from stdin")
+	scanner := bufio.NewScanner(os.Stdin)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line != "" {
+			e.parseLogLine(line)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Printf("Error reading from stdin: %v", err)
+	}
+}
+
+func init() {
+	// Register Prometheus metrics
+	prometheus.MustRegister(squidHitRatio)
+	prometheus.MustRegister(squidHitTotal)
+	prometheus.MustRegister(squidMissTotal)
+	prometheus.MustRegister(squidRequestsTotal)
+	prometheus.MustRegister(squidBytesTotal)
+	prometheus.MustRegister(squidResponseTime)
+}
+
+func main() {
+	flag.Parse()
+
+	log.Printf("Starting squid per-site exporter")
+	log.Printf("Listening on %s", *listenAddress)
+	log.Printf("Reading logs from stdin (use shell redirection for files)")
+
+	exporter := NewExporter()
+
+	// Start reading from stdin in background
+	go exporter.readFromStdin()
+
+	// Setup HTTP handlers
+	// Use HandlerFor with custom options to control content type format
+	handler := promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{
+		// Disable the escaping=values parameter to match expected format
+		EnableOpenMetrics: false,
+	})
+	http.Handle("/metrics", handler)
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`<html>
+			<head><title>Squid Per-Site Exporter</title></head>
+			<body>
+			<h1>Squid Per-Site Exporter</h1>
+			<p><a href='/metrics'>Metrics</a></p>
+			</body>
+			</html>`))
+	})
+
+	// Health check endpoint
+	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte("OK"))
+	})
+
+	// Start server; HTTPS required (no HTTP fallback)
+	if fileExists(*tlsCertFile) && fileExists(*tlsKeyFile) {
+		log.Printf("Starting HTTPS server on %s", *listenAddress)
+		log.Printf("Using TLS cert: %s", *tlsCertFile)
+		log.Printf("Using TLS key: %s", *tlsKeyFile)
+		log.Fatal(http.ListenAndServeTLS(*listenAddress, *tlsCertFile, *tlsKeyFile, nil))
+	}
+	log.Fatalf("TLS certificate or key not found (cert: %s, key: %s). HTTPS is required for metrics/health endpoints.", *tlsCertFile, *tlsKeyFile)
```

Comment:
This section should consult the value of `tlsRequired` and adjust the server start command correspondingly.